core/src/main/java/org/elasticsearch/index/shard/IndexShardCreationException.java

```diff
@@ -19,16 +19,18 @@
 
 package org.elasticsearch.index.shard;
 
+import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.common.io.stream.StreamInput;
 
 import java.io.IOException;
 
 /**
  */
-public class IndexShardCreationException extends IndexShardException {
+public class IndexShardCreationException extends ElasticsearchException {
```

Comment:
Do we need this dedicated exception? It isn't really used anywhere and the message should be enough... maybe this one is another one to remove?